File: download_subset_precip.py

# subset nc files to our area of interest 
import xarray as xr 
import numpy as np 
import pandas as pd 

#Go to this website and download the pr_ files first: https://www.northwestknowledge.net/metdata/data/

lats, lons = [30, 31], [-88, -87] 

for year in range(1979, 2022):
    file = 'pr_%s.nc' % year
    file_out = 'basin_prcp_%s.csv' % year 

    nc = xr.open_dataset(local_dir + file) 

    data = nc['precipitation_amount']

    data = data.loc[dict(lon = slice(lons[0], lons[1]))]
    data = data.loc[dict(lat = slice(lats[1], lats[0]))]

    mean_precip = np.mean(data, axis = (1,2))

    out = pd.DataFrame({'day': mean_precip['day'].values, 'prcp': mean_precip.values})
    # np.savetxt(file_out, out, delimiter=',')
    out.to_csv(file_out, index = False) 

File: download_subset_precip.R
library(dplyr)
library(ncdf4)
setwd("C:\\Users\\a_colleague\\Desktop\\basin_prcp_scripts\\data")

#Go to this website and download the pr_ files first: https://www.northwestknowledge.net/metdata/data/

pfiles <- list.files(pattern = "*.nc")

for(i in 1:42){
  year <- i + 1978
  nc_file <- nc_open(pfiles[i])
  #I figured out that these cells cover our basin
  prcp_3d <- ncvar_get(nc_file, varid = 'precipitation_amount', start = c(155, 229, 1),
                       count = c(10,15, -1))
  prcp_avg <- apply(prcp_3d, MARGIN = 3, FUN = mean)
  write.csv(file = paste('basin_prcp_', year, '.csv', sep = ''), x = prcp_avg)
}
